backend/src/integration-test/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/SpecificationControllerIntegrationTest.kt

```diff
@@ -121,13 +116,15 @@ internal class SpecificationControllerIntegrationTest {
             .andExpect(jsonPath("$.versions[0].metadata.version", equalTo("v2")))
 
         mockMvc.perform(
-            get("/api/v1/specifications/b6b06513-d259-4faf-b34b-a216b3daad6a")
+            get("/api/v1/specifications/unique-identifier")
                 .with(user("user"))
                 .with(csrf())
         )
             .andExpect(jsonPath("$.title", equalTo("Spec1")))
             .andExpect(jsonPath("$.versions[0].metadata.version", equalTo("v2")))
+            .andExpect(jsonPath("$.versions[0].metadata.title", equalTo("Spec1")))
```

Comment:
Yes, if I understand you right, this is the order the specifications would be displayed in the `specification-overview`, by upload date. Which reminds me of a related question: do we ever want to override this, and allow a service maintainer to sort by the version string? I thought it might be useful. For example if someone is uploading a large number of existing specifications at once, and doesn't want to take the time to manually sort.

Reply:
Yes and no: It should always be sorted by the version, and not be able to be overridden.

Reply:
Alphabetically sorted? It currently isn't: `@OrderBy("created DESC") val versions: List<VersionEntity>`

Reply:
Hm. Ideally, by version numbers, so that 4.1.0.RELEASE is better than 4.1.0.RC2 and also 4.1.0.BETA. But let's keep this out of scope for now. I'd suggest alphabetically descending, that should handle all the numbers well enough.

Another thing: Do we actually send all the versions on the first load? Because I think we should only the send the most current one and let it reload the other ones later, if the user actually wants them.